### ISSUE TITLE
Do not use .exists()

### ISSUE THREAD

**Issue by alexdashkov** [Volentix/venue-server#161]
I see a lot of examples in the code when you use `.exists` method. 

For example:

```python
user_check = User.objects.filter(
    email__iexact=data.get('email').strip()
)
if user_check.exists():
    try:
        user_profile = UserProfile.objects.get(
            user=user_check.last()
        )
```

Please, do not use this method if you plan to get data in the future. When you run it you execute 2 database queries instead of only one. If user object doesn't exist you will get an empty DB query that cost nothing for the DB, but if a user exists you will get only query instead of two. For example this part of a code you can rewrite like this: 

```python
user_check = User.objects.filter(
    email__iexact=data.get('email').strip()
).last() #  None if there is no user
# and here all the verification you have 
```
Also in this part you can see that you execute two query for the User model and one for user profile, but you can make only one like this: 
```sql 
SELECT "venue_userprofile"."email_confirmed"
FROM "auth_user"
       LEFT JOIN "venue_userprofile" ON ("auth_user"."id" = "venue_userprofile"."user_id")
WHERE UPPER("auth_user"."email" :: TEXT) = UPPER(SOME email)
```
Using ORM query like this one: 
```python
email_confirmed = User.objects.filter(
     email__iexact=data.get('email').strip()
).values('profiles__email_confirmed').last()
response['email_exists'] = email_confirmed['profiles__email_confirmed'] if email_confirmed else False
```

P.S. `exists` works very well but when you don't want to use data from the query(only check if exists).

**Comment by joemarct** [comment]
@alexdashkov I completely agree.

There are lots of other places where database queries can be further optimized. I'll create a branch `db-query-optimization` and will have  these`.exists()` refactoring as initial commit. I'll also create a more generic issue on this that you can refer to on your commits.

**Comment by joemarct** [comment]
Addressed in commit b5998ff037c21ab070e44f09d430247ab71440b4.

I simplified your suggested query further to:

```python
user_check = UserProfile.objects.filter(
    user__email__iexact=data.get('email').strip(),
    email_confirmed=True
)
if user_check:
    response['email_exists'] = True
```

**Comment by alexdashkov** [comment]
Btw in your last version ‘exists’ works pretty well because you don’t have
to get the result but only check that result exists.
